src/com/scottyplunkett/server/HTTPRequest.java

```diff
@@ -5,20 +5,27 @@
 import java.util.ArrayList;
 
 class HTTPRequest {
-    static ArrayList<String> read(BufferedReader in) throws IOException {
+    private ArrayList<String> read;
+    private String requestLine;
+
+    HTTPRequest(BufferedReader in) throws IOException {
+        read = new ArrayList<>();
+        read(in);
```

Comment:
The name collision here (`read` the method, `read` the private list) was a little confusing for me.

Reply:
I could see how that could be confusing. My goal was to create classes with phrasing that represented both the noun (the Value), and the verb (the Process) for ease of understanding.
ex.
Issue the command: Scott, read (verb) the book. 
Now get the value: Scott, give me your read. (noun).
ex. 
Scott, build (action) the building. 
Scott, show us the build (value).

I can see how this is less clarifying than I intended, and if it's confusing at first glance than I should change this templating syntax to be more expressive of the implementation as you've suggested below.

Reply:
I follow you - I think the problem here is that I had two concepts in my head.  The overloading or shadowing of a term is something a lot of languages handle differently.  So when I looked at it, for a minute, I was wondering, "Is this a special operation that a `String` has in Java, where you can call it like a function?"